src/telegram/process_update_task.rs

```diff
@@ -25,15 +26,90 @@ impl ProcessUpdateTask {
         Self { update }
     }
 
-    pub async fn process(&self) {
-        // TODO: process users messages here
-        // Create a ApiClient
+    pub async fn process(&self) -> Result<(), BotError> {
+        log::info!("Received a message {:?}", self.update);
+
         let api = AsyncApi::new(&RUST_TELEGRAM_BOT_TOKEN);
-        let opwm_token = &OPEN_WEATHER_MAP_API_TOKEN;
+        // for now with _ to pass warnings
+        let _opwm_token = &OPEN_WEATHER_MAP_API_TOKEN;
         let keypair = Rsa::private_key_from_pem(&BINARY_FILE).unwrap();
         let keypair = PKey::from_rsa(keypair).unwrap();
 
-        log::info!("Received a message {:?}", self.update);
+        if let UpdateContent::Message(message) = &self.update.content {
+            let (chat_id, user_id, user) = Self::get_info_from_message(message);
+
+            Self::send_typing(message, &api).await?;
+
+            let state = Self::fetch_state(&chat_id, user_id, user.clone(), &keypair).await?;
+
+            match state {
+                ClientState::Initial => match message.text.as_deref() {
+                    Some("/start") | Some("/start@RustWeather77Bot") => {
+                        Self::start(message, &user, &api).await?;
+                    }
+                    _ => {}
+                },
+
+                ClientState::Pattern => match message.text.as_deref() {
+                    Some("/pattern") | Some("/pattern@RustWeather77Bot") => {}
+                    _ => {}
+                },
+                _ => {}
+            }
+        }
+        Ok(())
+    }
+
+    pub async fn fetch_state(
+        chat_id: &i64,
+        user_id: u64,
+        user: String,
+        keypair: &PKey<Private>,
+    ) -> Result<ClientState, BotError> {
+        // Maybe here can be recycled pool from AsyncQueue from Fang for now this is fine
+        let db_controller = DbController::new().await?;
+        let state: ClientState = if !db_controller.check_user_exists(chat_id, user_id).await? {
+            db_controller
+                .insert_client(chat_id, user_id, user, keypair)
+                .await?;
+
+            ClientState::Initial
+        } else {
+            db_controller.get_client_state(chat_id, user_id).await?
+        };
+
+        Ok(state)
+    }
+
+    fn get_info_from_message(message: &Message) -> (i64, u64, String) {
+        let chat_id: i64 = message.chat.id;
+        let user_id: u64 = match message.from.as_deref() {
```

Comment:
user is always not empty in updates. so you can use unwrap() safely

the same comment is for `from`

Reply:
`expect` use